internal/config/loader_test.go

```diff
@@ -39,3 +39,475 @@ func TestLoadConfiguration(t *testing.T) {
 	}
 
 }
+
+func TestFullLoadConfiguration(t *testing.T) {
+	a := []string{"-config", "../../testdata/test.full.conf"}
+	// it should not error if config path is not set
+	err := Load("trickster-test", "0", a)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Test Proxy Server
+	if ProxyServer.ListenPort != 57821 {
+		t.Errorf("expected 57821, got %d", ProxyServer.ListenPort)
+	}
+
+	if ProxyServer.ListenAddress != "test" {
+		t.Errorf("expected test, got %s", ProxyServer.ListenAddress)
+	}
+
+	// Test Metrics Server
+	if Metrics.ListenPort != 57822 {
+		t.Errorf("expected 57821, got %d", Metrics.ListenPort)
```

Comment:
57822?

Reply:
Oh you set constants in the next Commit. NVM.

Reply:
Actually it's not the constants that makes this work, but rather the configuration files in `testdata/test.full.conf` and `testdata/test.empty.conf`. This test loads those files and expects them to resolve specific values.